YamlClientFailoverConfigBuilderTest test fails with JDK < 8

_master_ (commit f8ee8f31d151fcd6ed10e2bee6ee00ff6851f205)

Failed on ZingJDK7: http://jenkins.hazelcast.com/view/Official%20Builds/job/Hazelcast-3.x-ZingJDK7/21/testReport/

Stacktrace:
```
com.hazelcast.config.InvalidConfigurationException: Invalid YAML configuration
	at com.hazelcast.client.config.YamlClientFailoverConfigBuilder.parseAndBuildConfig(YamlClientFailoverConfigBuilder.java:140)
	at com.hazelcast.client.config.YamlClientFailoverConfigBuilder.build(YamlClientFailoverConfigBuilder.java:127)
	at com.hazelcast.client.config.YamlClientFailoverConfigBuilder.build(YamlClientFailoverConfigBuilder.java:117)
	at com.hazelcast.client.config.YamlClientFailoverConfigBuilderTest.init(YamlClientFailoverConfigBuilderTest.java:47)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:57)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:606)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:24)
	at com.hazelcast.test.AbstractHazelcastClassRunner$ThreadDumpAwareRunAfters.evaluate(AbstractHazelcastClassRunner.java:386)
	at org.junit.rules.ExpectedException$ExpectedExceptionStatement.evaluate(ExpectedException.java:239)
	at org.junit.rules.RunRules.evaluate(RunRules.java:20)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:325)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:78)
	at com.hazelcast.test.HazelcastSerialClassRunner.runChild(HazelcastSerialClassRunner.java:50)
	at com.hazelcast.test.HazelcastSerialClassRunner.runChild(HazelcastSerialClassRunner.java:29)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
	at com.hazelcast.test.AbstractHazelcastClassRunner$1.evaluate(AbstractHazelcastClassRunner.java:302)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
	at org.junit.runners.Suite.runChild(Suite.java:128)
	at org.junit.runners.Suite.runChild(Suite.java:27)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
	at org.apache.maven.surefire.junitcore.JUnitCore.run(JUnitCore.java:55)
	at org.apache.maven.surefire.junitcore.JUnitCoreWrapper.createRequestAndRun(JUnitCoreWrapper.java:137)
	at org.apache.maven.surefire.junitcore.JUnitCoreWrapper.executeEager(JUnitCoreWrapper.java:107)
	at org.apache.maven.surefire.junitcore.JUnitCoreWrapper.execute(JUnitCoreWrapper.java:83)
	at org.apache.maven.surefire.junitcore.JUnitCoreWrapper.execute(JUnitCoreWrapper.java:75)
	at org.apache.maven.surefire.junitcore.JUnitCoreProvider.invoke(JUnitCoreProvider.java:158)
	at org.apache.maven.surefire.booter.ForkedBooter.invokeProviderInSameClassLoader(ForkedBooter.java:379)
	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:340)
	at org.apache.maven.surefire.booter.ForkedBooter.execute(ForkedBooter.java:125)
	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:413)
Caused by: com.hazelcast.internal.yaml.YamlException: An error occurred while loading and parsing the YAML stream
	at com.hazelcast.internal.yaml.YamlLoader.load(YamlLoader.java:74)
	at com.hazelcast.client.config.YamlClientFailoverConfigBuilder.parseAndBuildConfig(YamlClientFailoverConfigBuilder.java:138)
	... 43 more
Caused by: java.lang.UnsupportedOperationException: Processing YAML documents requires Java 8 or higher version
	at com.hazelcast.internal.yaml.YamlUtil.ensureRunningOnJava8OrHigher(YamlUtil.java:116)
	at com.hazelcast.internal.yaml.YamlLoader.getLoad(YamlLoader.java:157)
	at com.hazelcast.internal.yaml.YamlLoader.load(YamlLoader.java:69)
	... 44 more
```

--- a/hazelcast-client/src/test/java/com/hazelcast/client/config/YamlClientFailoverConfigBuilderTest.java
+++ b/hazelcast-client/src/test/java/com/hazelcast/client/config/YamlClientFailoverConfigBuilderTest.java
@@ -31,6 +31,7 @@ import java.io.ByteArrayInputStream;
 import java.net.URL;
 import java.util.Properties;
 
+import static com.hazelcast.test.HazelcastTestSupport.assumeThatJDK8OrHigher;
 import static org.junit.Assert.assertEquals;
 
 @RunWith(HazelcastSerialClassRunner.class)
@@ -42,6 +43,8 @@ public class YamlClientFailoverConfigBuilderTest extends AbstractClientFailoverC
 
     @Before
     public void init() throws Exception {
+        assumeThatJDK8OrHigher();
+
         URL schemaResource = YamlClientFailoverConfigBuilderTest.class.getClassLoader()
                                                                       .getResource("hazelcast-client-failover-sample.yaml");
         fullClientConfig = new YamlClientFailoverConfigBuilder(schemaResource).build();

